### DISCUSSION TITLE
Can Teleport be used with Tailscale to secure SSH access without exposure to the internet?

### DISCUSSION THREAD

**Discussion by ahelwer** [gravitational/teleport#20245] [Q&A] [answered]
I have a few computers scattered around my home that I like to SSH into, either while inside my LAN or elsewhere. Currently I use -sk U2F SSH keys for this purpose, plus Tailscale so I can access the computers wherever. However, I much prefer Teleport's cert-based auth so I don't have to worry about private/public key management. I have the following questions:

1. Is it possible to run the Teleport cluster management node inside a Tailscale VPN, or does it have to be exposed to the internet so it can be associated with a domain name and get a certificate from LetsEncrypt?
1. Is all SSH traffic proxied through the Teleport proxy server, or only reverse-tunneling connections?
1. If all nodes (including the Teleport cluster management node) are part of a Tailscale VPN, you don't need to use the reverse-tunneling feature, right?

I certainly could set up a publicly-accessible Teleport cluster management node on Oracle or Azure without cost to myself, but worry that would be exposing my personal network to greater risk than exists now by making it so if that public node is compromised an attacker would instantly gain access to my entire network. For me it would also be preferable to not proxy all SSH traffic through the Teleport proxy, since this is just for personal use so I don't need to worry about auditing or real-time session termination. Thanks for any help you all can provide!

**Comment by webvictim** [comment, marked as answer]
1. I haven't tried, but I know colleagues have - it's possible to run the Teleport proxy server behind a Tailscale VPN. You'll need to figure out a way to make DNS and certs work, or you can just use IP addresses and maybe `teleport start --insecure` on your agent services.

2. All traffic hits the Teleport proxy server, but if your agents are joined to the Teleport auth server (port 3025) then the connections to them will be direct to the IP on port 3022. This is only relevant to the Teleport `ssh_service`. In all other cases where reverse tunnels are used (with agents joined to the Teleport proxy server on port 443 or 3080 with a reverse tunnel on 3024) all traffic always goes through the proxy server.

3. Theoretically you don't need to use reverse tunnelling if you only want SSH access to the nodes, correct. My recommendation would be to use reverse tunnelling simply for ease, but if all your nodes have Tailscale connectivity and unique addresses, you could set their `advertise_ip` to the Tailscale IP and get the same effect.

**Reply by ahelwer** [reply]
Thanks for the detailed answer! Just to clarify your answer to question 2, so if I tsh into a node that is directly accessible on my network will all back-and-forth data flow through the teleport proxy or is there some kind of handoff so I can ssh directly in to the node without touching the teleport proxy after authorization?

**Reply by webvictim** [reply]
It's the latter. If the Teleport `ssh_service` on your node is listening on port 3022, your client will connect directly to that hostname/port and traffic won't go through the Teleport auth/proxy in that scenario.

**Reply by tejasri354970** [reply]
HI @webvictim 
Could you please explain how they configured the Teleport proxy server behind the Open  VPN?
 I'm trying the configure the setting but I'm getting errors like
 502 Bad gateway error.
 Could you please help me out this one .
